File: engine/tanaguru-api/src/main/java/org/tanaguru/entity/audit/Audit.java
/*
 * Tanaguru - Automated webpage assessment
 * Copyright (C) 2008-2015  Tanaguru.org
 *
 * This file is part of Tanaguru.
 *
 * Tanaguru is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * Contact us by mail: tanaguru AT tanaguru DOT org
 */
package org.tanaguru.entity.audit;

import java.util.Date;

import org.tanaguru.sdk.entity.Entity;
import java.util.Collection;
import org.tanaguru.entity.parameterization.Parameter;
import org.tanaguru.entity.reference.Test;
import org.tanaguru.entity.subject.WebResource;

/**
 * 
 * @author jkowalczyk
 */
public interface Audit extends Entity {

    /**
     *
     * @param contentList
     *            the content list to add
     */
    void addAllContent(Collection<Content> contentList);

    /**
     *
     * @param grossResultList
     *            the gross result list to add
     */
    void addAllGrossResult(Collection<ProcessResult> grossResultList);

    /**
     *
     * @param netResultList
     *            the net result list to add
     */
    void addAllNetResult(Collection<ProcessResult> netResultList);

    /**
     *
     * @param testList
     *            the test set to add
     */
    void addAllTest(Collection<Test> testList);

    /**
     *
     * @param content
     *            the content to add
     */
    void addContent(Content content);

    /**
     *
     * @param grossResult
     *            the gross result to add
     */
    void addGrossResult(ProcessResult grossResult);

    /**
     *
     * @param netResult
     *            the net result to add
     */
    void addNetResult(ProcessResult netResult);

    /**
     *
     * @param subject
     *            the subject to set
     */
    void setSubject(WebResource subject);

    /**
     *
     * @param test
     *            the test to add
     */
    void addTest(Test test);

    /**
     *
     * @return the comment
     */
    String getComment();

    /**
     *
     * @return the content list
     */
    Collection<Content> getContentList();

    /**
     *
     * @return the date of creation
     */
    Date getDateOfCreation();
    
    /**
    *
    * @return the date of creation
    */
    Date getManualAuditDateOfCreation();

    /**
     *
     * @return the gross result list
     */
    Collection<ProcessResult> getGrossResultList();

    /**
     *
     * @return the net result list
     */
    Collection<ProcessResult> getNetResultList();

    /**
     *
     * @return the status
     */
    AuditStatus getStatus();

    /**
     *
     * @return the subject
     */
    WebResource getSubject();

    /**
     *
     * @return the test list
     */
    Collection<Test> getTestList();

    /**
     *
     * @param comment
     *            the comment to set
     */
    void setComment(String comment);

    /**
     *
     * @param contentList
     *            the content list to set
     */
    void setContentList(Collection<Content> contentList);

    /**
     *
     * @param manualAuditDateOfCreation
     *            the date of manual audit launch to set
     */
    void setManualAuditDateOfCreation(Date manualAuditDateOfCreation);
    
    /**
    *
    * @param dateOfCreaction
    *            the date of content loading to set
    */
    void setDateOfCreation(Date dateOfCreaction);

    /**
     *
     * @param grossResultList
     *            the gross result list to set
     */
    void setGrossResultList(Collection<ProcessResult> grossResultList);

    /**
     *
     * @param netResultList
     *            the net result list to set
     */
    void setNetResultList(Collection<ProcessResult> netResultList);

    /**
     *
     * @param status
     *            the status to set
     */
    void setStatus(AuditStatus status);

    /**
     *
     * @param testList
     *            the test list to set
     */
    void setTestList(Collection<Test> testList);

    /**
     * 
     * @param parameterSet
     */
    void setParameterSet(Collection<Parameter> parameterSet);

    /**
     * 
     * @return
     */
    Collection<Parameter> getParameterSet();

    /**
     * 
     * @param parameter
     */
    void addParameter(Parameter parameter);
}

File: server/server-model/src/main/kotlin/org/asqatasun/model/Audit.kt
/*
 *  Asqatasun - Automated webpage assessment
 *  Copyright (C) 2008-2020  Asqatasun.org
 *
 *  This file is part of Asqatasun.
 *
 *  Asqatasun is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Affero General Public License as
 *  published by the Free Software Foundation, either version 3 of the
 *  License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Affero General Public License for more details.
 *
 *  You should have received a copy of the GNU Affero General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 *  Contact us by mail: asqatasun AT asqatasun DOT org
 */
package org.asqatasun.model

import io.swagger.v3.oas.annotations.media.Schema
import org.asqatasun.entity.audit.Audit
import org.asqatasun.entity.audit.AuditStatus
import java.util.*

@Schema(
    title = "Audit",
    description = "The audit with its characteristics"
)
data class AuditDto(
    @field:Schema(
        description = "Unique id of the audit",
        example = "42",
    )
    val id: Long,
    // WebResource already documented in Subject.kt
    val subject: WebResourceDto,
    // AuditStatus already described in engine/asqatasun-api/src/main/java/org/asqatasun/entity/audit/AuditStatus.java
    val status: AuditStatus,
    val date: Date,
    @field:Schema(
        description = "Tag(s) associated to the audit",
        example = "monthly_audit_on_localgovs, localgovs_France, 2022-01-01",
        type = "List<String>",
    )
    val tags: List<String>,
    // Referential already documented in Reference.kt
    val referential: Referential,
    @field:Schema(
        description = "Referential level",
        example = "AA",
        type = "String",
    )
    val referentialLevel: String
)

fun Audit.toAuditDto(webResource: WebResourceDto, tags: List<String>, rawParameterValue: String) = AuditDto(
    id,
    webResource,
    status,
    dateOfCreation,
    tags,
    Referential.fromCode(rawParameterValue.split(';')[0])!!,
    rawParameterValue.split(';')[1]
)

